src/shell/task.c

```diff
@@ -228,6 +260,38 @@ int shell_task_kill (struct shell_task *task, int signum)
     return rc;
 }
 
+/*  flux_shell_task_t API implementation:
+ */
+
+flux_cmd_t *flux_shell_task_cmd (flux_shell_task_t *task)
+{
+    return task->cmd;
+}
+
+flux_subprocess_t *flux_shell_task_subprocess (flux_shell_task_t *task)
+{
+    return task->proc;
+}
+
+int flux_shell_task_channel_subscribe (flux_shell_task_t *task,
+                                       const char *name,
+                                       flux_shell_task_io_f cb,
+                                       void *arg)
+{
+    struct channel_watcher *cw;
+
+    if (zhashx_lookup (task->subscribers, name)) {
+        errno = EEXIST;
+        return -1;
+    }
+    if (!(cw = calloc (1, sizeof (*cw))))
+        return -1;
+    cw->cb = cb;
+    cw->arg = arg;
+    zhashx_update (task->subscribers, name, cw);
```

Comment:
I'm tempted since it is the "right" thing to do. I was missing the correct flags to export dynamic symbols from flux-shell anyway. Seems the "right" thing to do would be to perform both of these steps once shell plugins are ready.